packages/date-picker/src/vaadin-date-picker-mixin.js

```diff
@@ -411,6 +414,29 @@ export const DatePickerMixin = (subclass) =>
       );
 
       this.addController(new VirtualKeyboardController(this));
+
+      this.$.overlay.addEventListener('opened-changed', (e) => {
```

Comment:
**nit:** These two overlay listeners could be added to the template instead.

Reply:
I wanted to avoid adding same listeners also in `vaadin-date-picker-light`, so using a mixin here.

Reply:
OK, my concern is that it becomes not very consistent now considering that there are overlay listeners that are defined in both regular and light component template:

```js
on-vaadin-overlay-open="_onOverlayOpened"
on-vaadin-overlay-closing="_onOverlayClosed"
```

Not a big deal though.

Reply:
Actually, we can get rid of the `opened-changed` listener completely: it's no longer needed after removing usage of `disable-upgrade`. Updated to use two-way binding for `opened` property, as in the light version.

Also moved the `vaadin-overlay-escape-press` listener to the template for consistency.

Reply:
Excellent!